plugins/spu2-x/src/wx/wxConfig.cpp

```diff
@@ -18,471 +18,486 @@
-    m_top_box->Layout();
-    SetSizerAndFit(m_top_box);
+void DebugTab::Load()
+{
+	debug_check->SetValue(DebugEnabled);
+
+	show_check->SetValue(_MsgToConsole);
+	key_check->SetValue(_MsgKeyOnOff);
+	voice_check->SetValue(_MsgVoiceOff);
+	dma_check->SetValue(_MsgDMA);
+	autodma_check->SetValue(_MsgAutoDMA);
+	buffer_check->SetValue(_MsgOverruns);
+	adpcm_check->SetValue(_MsgCache);
+
+	dma_actions_check->SetValue(_AccessLog);
+	dma_writes_check->SetValue(_DMALog);
+	auto_output_check->SetValue(_WaveLog);
+
+	core_voice_check->SetValue(_CoresDump);
+	memory_check->SetValue(_MemDump);
+	register_check->SetValue(_RegDump);
+	Update();
+}
 
-    launch_debug_dialog->Enable(debug_check->GetValue());
-    launch_adv_dialog->Enable(m_sync_select->GetCurrentSelection() == 0);
+void DebugTab::Save()
+{
+	DebugEnabled = debug_check->GetValue();
+
+	_MsgToConsole = show_check->GetValue();
+	_MsgKeyOnOff = key_check->GetValue();
+	_MsgVoiceOff = voice_check->GetValue();
+	_MsgDMA = dma_check->GetValue();
+	_MsgAutoDMA = autodma_check->GetValue();
+	_MsgOverruns = buffer_check->GetValue();
+	_MsgCache = adpcm_check->GetValue();
+
+	_AccessLog = dma_actions_check->GetValue();
+	_DMALog = dma_writes_check->GetValue();
+	_WaveLog = auto_output_check->GetValue();
+
+	_CoresDump = core_voice_check->GetValue();
+	_MemDump = memory_check->GetValue();
+	_RegDump = register_check->GetValue();
 }
 
-void Dialog::CallReconfigure(wxCommandEvent &event)
+void DebugTab::Update()
 {
-    Reconfigure();
+	if (debug_check->GetValue())
+	{
+		show_check->Enable();
+
+		key_check->Enable();
+		voice_check->Enable();
+		dma_check->Enable();
+		autodma_check->Enable();
+		buffer_check->Enable();
+		adpcm_check->Enable();
+
+		dma_actions_check->Enable();
+		dma_writes_check->Enable();
+		auto_output_check->Enable();
+
+		core_voice_check->Enable();
+		memory_check->Enable();
+		register_check->Enable();
+	}
+	else
+	{
+		show_check->Disable();
+
+		key_check->Disable();
+		voice_check->Disable();
+		dma_check->Disable();
+		autodma_check->Disable();
+		buffer_check->Disable();
+		adpcm_check->Disable();
+
+		dma_actions_check->Disable();
+		dma_writes_check->Disable();
+		auto_output_check->Disable();
+
+		core_voice_check->Disable();
+		memory_check->Disable();
+		register_check->Disable();
+	}
 }
 
-void Dialog::OnButtonClicked(wxCommandEvent &event)
+void DebugTab::CallUpdate(wxCommandEvent& /*event*/)
 {
-    wxButton *bt = (wxButton *)event.GetEventObject();
-
-    if (bt == launch_debug_dialog)
-    {
-        auto debug_dialog = new DebugDialog;
-        debug_dialog->Display();
-        wxDELETE(debug_dialog);
-    }
-
-    if (bt == launch_adv_dialog)
-    {
-        auto adv_dialog = new SoundtouchCfg::AdvDialog;
-        adv_dialog->Display();
-        wxDELETE(adv_dialog);
-    }
+	Update();
 }
 
-void Dialog::ResetToValues()
+Dialog::Dialog()
+	: wxDialog(nullptr, wxID_ANY, "SPU2-X Config", wxDefaultPosition, wxDefaultSize, wxCAPTION | wxCLOSE_BOX)
```

Comment:
If we are getting rid of the gsdx branding then we might as well update this one too.